### PR TITLE
add checks for nodes label

### PR DESCRIPTION
**Pull request by BlueShells**
sometimes a node lost some label causing us problem

### CODE REVIEW

File: scripts/monitoring/cron-send-node-labels-status.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    args = parser.parse_args()
+
+    if args.verbose > 0:
+        logger.setLevel(logging.INFO)
+    if args.verbose > 1:
+        logger.setLevel(logging.DEBUG)
+
+    return args
+
+def check_lable_status():
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-node-labels-status.py]: label*

File: scripts/monitoring/cron-send-node-labels-status.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        if len(ip_status) < 5:
+            logger.info('this node lack of ip: %s', item['metadata']['name'])
+            result_ip = result_ip + 1
+        # if the label on the host less than 11 , means lack of label, for 3.9
+        if len(labels) < 11:
+            logger.info('this node lack of labels: %s', item['metadata']['name'])
+            result_status = result_status  + 1
+
+    #if everything fine , result_ip and result_status both should 0 , if not 0 , means something wrong
+    return result_ip + result_status
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-node-labels-status.py]: should we seperate the number of "address" faults vs "label" faults?

File: scripts/monitoring/cron-send-node-labels-status.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    label_info = runOCcmd_yaml("get node ")
+    for item in label_info['items']:
+        labels = item['metadata']['labels']
+        ip_status = item['status']['addresses']
+        # if the host have less than 5 address , means node miss ip
+        if len(ip_status) < 5:
+            logger.info('this node lack of ip: %s', item['metadata']['name'])
+            result_ip = result_ip + 1
+        # if the label on the host less than 11 , means lack of label, for 3.9
+        if len(labels) < 11:
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-node-labels-status.py]: would it be better to check if specific labels are available? I'm not 100% sure on which ones, but relying on a count may not work. If the team adds new labels, but important ones are still missing, then this check will still pass.

File: scripts/monitoring/cron-send-node-labels-status.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+def check_lable_status():
+    """get all the info of all node """
+    result_ip = 0
+    result_status = 0
+    label_info = runOCcmd_yaml("get node ")
+    for item in label_info['items']:
+        labels = item['metadata']['labels']
+        ip_status = item['status']['addresses']
+        # if the host have less than 5 address , means node miss ip
+        if len(ip_status) < 5:
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-node-labels-status.py]: should we check specific keys for this? I don't hink a count is suitable.

File: scripts/monitoring/cron-send-node-labels-status.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            logger.info('this node lack of ip: %s', item['metadata']['name'])
+            result_ip = result_ip + 1
+        # if the label on the host less than 11 , means lack of label, for 3.9
+        if len(labels) < 11:
+            logger.info('this node lack of labels: %s', item['metadata']['name'])
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-node-labels-status.py]: we should say which labels are missing.

File: scripts/monitoring/cron-send-node-labels-status.py (hunk 3 above)
Comment (wanghaoran1988) [reply scripts/monitoring/cron-send-node-labels-status.py]: I'd like add checks like this:
1. If it has "type=master", we should check whether the node has "node-role.kubernetes.io/master=true"
2. If it has "type=infra", we should check whether the node has "node-role.kubernetes.io/infra=true"
3. If it has "type=compute", we should check whether the node has "node-role.kubernetes.io/compute=true"

File: scripts/monitoring/cron-send-node-labels-status.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    result = True
+    hostname = labels['hostname']
+    host_type = labels['type']
+    # if the node miss the hostname and type label
+    if hostname and host_type:
+        pass
+    else:
+        return False
+
+    master_labels = {'beta.kubernetes.io/arch':'amd64', 'beta.kubernetes.io/instance-type': 'm4.xlarge', 'beta.kubernetes.io/os': 'linux', 'failure-domain.beta.kubernetes.io/region': 'us-east-1', 'failure-domain.beta.kubernetes.io/zone': 'us-east-1a', 'hostname': 'tke-master-9f206', 'kubernetes.io/hostname': 'ip-10-147-203-48.ec2.internal', 'network.openshift.io/not-enforcing-egress-network-policy':'true', 'node-role.kubernetes.io/master': 'true', 'region': 'us-east-1', 'type': 'master'}
```
Comment (wanghaoran1988) [inline scripts/monitoring/cron-send-node-labels-status.py]: Do we need check all the labels? I don't think every node will have label 'region': 'us-east-1' .

**Comment by drewandersonnz** [comment]
+1

**Review by drewandersonnz — APPROVED** [review]

**Pull request merged**